### DISCUSSION TITLE
Definition of jaeger span followFrom

### DISCUSSION THREAD

**Discussion by SwanHtetAung** [open-telemetry/opentelemetry-specification#4072] [Q&A] [answered]
Hi everyone, I have been looking at the definition of [followFrom](https://github.com/open-telemetry/opentelemetry-go/blob/6a0fa3c911a061198302cd1972c2e1a9e5ffa87f/semconv/v1.25.0/trace.go#L77) defined here which says
`// The parent Span doesn't depend in any way on the result of the child Span`. 

So, looking at the current bridge layer, it seems like if span reference type is followFrom, it will start a new traceID and dropped all [existing baggage](https://github.com/open-telemetry/opentelemetry-go/blob/6a0fa3c911a061198302cd1972c2e1a9e5ffa87f/bridge/opentracing/bridge.go#L586), unless the parent context is also provided when creating a span.

My question is whether a Jaeger Span created using followFrom should still propagate Jaeger baggage. If it does, will the child span propagate the Jaeger baggage with a different traceID or the same traceID? Or does followFrom imply that Jaeger will completely drop everything and start a new trace?

I feel that the definition of followFrom is not clear enough. It just states that the parent does not depend on the child's outcome. Does this also imply that the child should not propagate the parent's baggage and traceID?

Defenition of jaeger Baggage -> https://www.jaegertracing.io/docs/1.57/architecture/#baggage

cc @yurishkuro

**Comment by yurishkuro** [comment]
I don't understand the question. Spans in Jaeger native data model (close to OpenTracing) do not have an explicit "parent", but one or more references. By convention a single reference is treated as a parent, regardless of the reference type. But the model also allows that single reference to point to a different trace ID. The trace ID is determined by how the span is created.

**Comment by SwanHtetAung** [comment]
So, If I just do ->
A :=`opentracing.StartSpan(
		         "",
			opentracing.FollowsFrom(parentSpan.Context()))`
Is it reasonable to assume that,span A will be a root span with a new traceID and new baggage, or can I assume span A will not be a root span and it's traceID will be a parentSpan traceID and it will still contain parentSpan baggage.

Like, Can I say that FollowsFrom just only tell us the current span do not depend on the parent Span, but does not tell whether it should be a new root span or not.

**Reply by yurishkuro** [reply]
> `opentracing.StartSpan`

a) opentracing is deprecated
b) the OpenTracing spec does not prescribe the exact behavior (trace ID is not part of OpenTracing spec, it's implementation detail)
c) if you use Jaeger SDK (also deprecated) that implements `opentracing.StartSpan` then it will **not** create a new trace ID, the parentSpan's trace-id will be used
d) if you use OTEL SDK with OT bridge, I assume it will behave similar to Jaeger SDK and reuse trace-id (but you can check the code for that)

**Reply by SwanHtetAung** [reply]
For d. I look at it, it does not reuse the trace-id. 
Here is the code link:
https://github.com/open-telemetry/opentelemetry-go/blob/6a0fa3c911a061198302cd1972c2e1a9e5ffa87f/bridge/opentracing/bridge.go#L586
It only honors the Child-OF span reference type.

Jaeger SDK does reuse the traceID but otel does not. So I am wondering what will be the correct standard I should go with.

**Reply by yurishkuro** [reply, marked as answer]
> So I am wondering what will be the correct standard I should go with.

I would avoid using opentracing altogether, go with OTEL API.

**Reply by SwanHtetAung** [reply]
Gotcha Thanks!

**Reply by SwanHtetAung** [reply]
Should followFrom definition be updated to mention that it will result in the creation of a new root span? Something like

// The parent Span doesn't depend in any way on the result of the **new root** child Span.
